homeassistant/components/reolink/views.py

```diff
@@ -0,0 +1,138 @@
+        )
+
+    async def get(
+        self,
+        request: web.Request,
+        config_entry_id: str,
+        channel: str,
+        stream_res: str,
+        vod_type: str,
+        filename: str,
+        retry: int = 2,
+    ) -> web.StreamResponse:
+        """Get playback proxy video response."""
+        retry = retry - 1
+
+        filename = filename.replace("|", "/")
+        ch = int(channel)
+        host = get_host(self.hass, config_entry_id)
+
+        try:
+            mime_type, reolink_url = await host.api.get_vod_source(
+                ch, filename, stream_res, VodRequestType(vod_type)
+            )
+        except ReolinkError as err:
+            _LOGGER.warning("Reolink playback proxy error: %s", str(err))
+            return web.Response(body=str(err), status=HTTPStatus.BAD_REQUEST)
+
+        if _LOGGER.isEnabledFor(logging.DEBUG):
+            _LOGGER.debug(
+                "Opening VOD stream from %s: %s",
+                host.api.camera_name(ch),
+                host.api.hide_password(reolink_url),
+            )
+
+        try:
+            reolink_response = await self.session.get(
+                reolink_url,
+                timeout=ClientTimeout(
+                    connect=15, sock_connect=15, sock_read=5, total=None
+                ),
+            )
+        except ClientError as err:
+            err_str = host.api.hide_password(
+                f"Reolink playback error while getting mp4: {err!s}"
+            )
+            if retry <= 0:
+                _LOGGER.warning(err_str)
+                return web.Response(body=err_str, status=HTTPStatus.BAD_REQUEST)
+            _LOGGER.debug("%s, renewing token", err_str)
+            await host.api.expire_session(unsubscribe=False)
+            return await self.get(
+                request, config_entry_id, channel, stream_res, vod_type, filename, retry
+            )
+
+        if reolink_response.content_type not in [
+            "video/mp4",
+            "apolication/octet-stream",
+        ]:
+            err_str = f"Reolink playback expected video/mp4 but got {reolink_response.content_type}"
+            _LOGGER.error(err_str)
+            return web.Response(body=err_str, status=HTTPStatus.BAD_REQUEST)
+
+        response = web.StreamResponse(
+            status=200,
+            reason="OK",
+            headers={
+                "Content-Type": "video/mp4",
+            },
+        )
+
+        if reolink_response.content_length is not None:
+            response.content_length = reolink_response.content_length
+
+        await response.prepare(request)
+
+        try:
+            async for chunk in reolink_response.content.iter_chunked(65536):
+                await response.write(chunk)
+        except TimeoutError:
+            _LOGGER.debug(
+                "Timeout while reading Reolink playback from %s, writing EOF",
+                host.api.nvr_name,
+            )
+
+        reolink_response.release()
```

Comment:
Should it be in a finally block?

Reply:
I don't think it is nessesary.
I dont even know if the release() is actually needed, I think aiohttp will clean it up by itself.
But since it is a rather large payload (few Mb) I think it is nice to release it as soon as possible to free resources.